src/Simple/JSON.purs

```diff
@@ -143,6 +166,12 @@ instance writeForeignArray :: WriteForeign a => WriteForeign (Array a) where
 instance writeForeignNullOrUndefined :: WriteForeign a => WriteForeign (NullOrUndefined a) where
   writeImpl (NullOrUndefined a) = maybe undefined writeImpl a
 
+instance writeForeignMaybe :: WriteForeign a => WriteForeign (Maybe a) where
+  writeImpl (Just a) = writeImpl a
+  writeImpl Nothing = null
```

Comment:
I'd use `undefined` for `Nothing` to get smaller serialisation.

Reply:
I think a lot of existing implementations use Maybe -> null encoding though :cry: 

I think just trying to be compatible with Aeson and other libs would be good enough for Maybe, even though I would much prefer undefined too

Reply:
Won't Aeson interpret a missing field as `Nothing`?

Reply:
Not afaict, it will just fail to decode if the field doesn't exist as `null`. But this is the encoding case, where writing `null` would match Aeson's `ToJSON` behavior, but yeah, I would prefer to write out undefined (missing fields in JSON) if this isn't a concern for @dwhitney

Reply:
Looks like they use different functions depending on whether a missing key should be valid or not:

https://github.com/bos/aeson/blob/3a778548a928a5bb94a3defb526c985a4016e161/Data/Aeson/Types/FromJSON.hs#L689

Reply:
encoding `Nothing` to `null` is what I've encountered in the json serialization libs I've used. I also use GraphQL quite a bit and optional values are given the value of `null` when encoding json in those responses. I'd call it "industry standard" but don't quote me ;)